### DISCUSSION TITLE
Listing all AUR packages

### DISCUSSION THREAD

**Discussion by sa-mendez** [fosskers/aura#727] [Q&A] [answered]
I know you can search for AUR packages by word.  But in aura, is there an AUR equivalent for listing all the standard repo packages ( **_aura -Sl_**) ?

**Comment by micwoj92** [comment, marked as answer]
`pacman -Qm` works with installed packages that are not found in repos, that's the closest that can come to my mind.

**Reply by sa-mendez** [reply]
Thanks.

Yeah, I am aware of that.   But what I was hoping for, was to get a list of packages in the AUR that are not necessarily installed locally.

**Reply by fosskers** [reply]
That's not currently possible, and you actually probably would want it anyway, haha, given how many thousands of AUR packages there are. 

There are actually such lists available, however, if you're interested in putting a little script together that tells you what you're looking for.

**Reply by sa-mendez** [reply]
> That's not currently possible, and you actually probably would want it anyway, haha, given how many thousands of AUR packages there are.

The motivation for doing this, is to use aura with a fuzzy finder (this is documented for pacman here: [FZF](https://wiki.archlinux.org/title/Fzf#Arch_specific_fzf_uses)).

You could  use the following with aura (which works for only the standard repos):
`aura -Slq | fzf --multi --preview 'aura -Si {1}' | xargs -ro sudo aura -S`

 I would like to be able to use a similar command that queries and uses the AUR, but in order to do so, aura would have to provide the capability to get a list of AUR packages.

**Reply by fosskers** [reply]
I could potentially provide a "heavy sync" that pulls the tarball of every AUR package, and use that to establish the local repo cache. That would only apply to Aura 4 though, the current WIP.

**Reply by sa-mendez** [reply]
If I understand you correctly, the "heavy sync" operation you describe might be overkill. If you got to the following url: [https://aur.archlinux.org/packages.gz](https://aur.archlinux.org/packages.gz)

It gives you a gzip of all of the AUR packages as a text file, that can then be used to generate the list of all AUR packages.

Making this part of Aura 4 would be really appreciated.   If you think the effort on this is not to too great, I would even volunteer to submit the PR for this feature with the hope of making it a part of aura sooner.  I recently started learning Haskell (its one of the reasons this project appealed to me) and this might be something I can work on.

**Reply by micwoj92** [reply]
Aura 4 is rewrite in Rust.

**Reply by sa-mendez** [reply]
Oh I see.

**Reply by fosskers** [reply]
Yes that's precisely the tarball I was talking about. It could be used to prepopulate Aura's local cache of AUR packages.

**Reply by sa-mendez** [reply]
Understood.  Thank you
